src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Main.cs

```diff
@@ -0,0 +1,319 @@
+
+                foreach (var altName in found.AltNames)
+                {
+                    if (altName.Contains(query.Search, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Return a list context menu entries for a given <see cref="Result"/> (shown at the right side of the result)
+        /// </summary>
+        /// <param name="selectedResult">The <see cref="Result"/> for the list with context menu entries</param>
+        /// <returns>A list context menu entries</returns>
+        public List<ContextMenuResult> LoadContextMenus(Result selectedResult)
+        {
+            return ContextMenuHelper.GetContextMenu(selectedResult, _assemblyName);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Wrapper method for <see cref="Dispose"/> that dispose additional objects and events form the plugin itself
+        /// </summary>
+        /// <param name="disposing">Indicate that the plugin is disposed</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed || !disposing)
+            {
+                return;
+            }
+
+            if (!(_context is null))
+            {
+                _context.API.ThemeChanged -= OnThemeChanged;
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string GetTranslatedPluginTitle()
+        {
+            return Name;
+        }
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string GetTranslatedPluginDescription()
+        {
+            return Description;
+        }
+
+        /// <summary>
+        /// Change all theme-based elements (typical called when the plugin theme has changed)
+        /// </summary>
+        /// <param name="oldtheme">The old <see cref="Theme"/></param>
+        /// <param name="newTheme">The new <see cref="Theme"/></param>
+        private void OnThemeChanged(Theme oldtheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+        }
+
+        /// <summary>
+        /// Update all icons (typical called when the plugin theme has changed)
+        /// </summary>
+        /// <param name="theme">The new <see cref="Theme"/> for the icons</param>
+        private void UpdateIconPath(Theme theme)
+        {
+            _defaultIconPath = theme == Theme.Light || theme == Theme.HighContrastWhite
+                ? _lightSymbol
+                : _darkSymbol;
+        }
+
+        /// <summary>
+        /// Translate all setting entires
+        /// </summary>
+        private void TranslateAllSettings()
```

Comment:
Let's move into helper class for better code structure: `CreateSettingsList`

Reply:
move what into own class?  this is touching internal class vars.

Reply:
The translation method. (See my general comment fir the review.)

Reply:
put into separate class